app/controllers/accounts_controller.rb

```diff
@@ -117,41 +120,62 @@ def update
     account = current_organization.accounts.find(params[:id])
     return forbidden unless account && current_account.allowed_to_edit_account?(account, current_organization)
     return json(account) unless account.update_attributes pick(params, :first_name, :last_name, :email,:language, :document_language)
+
     role = pick(params, :role)
     #account.update_attributes(role) if !role.empty? && current_account.admin?
     membership = current_organization.role_of(account)
     membership.update_attributes(role) if !role.empty? && current_account.admin?
     password = pick(params, :password)[:password]
     if (current_account.id == account.id) && password
-      account.password = password
-      account.save
+      account.update(password: password)
     end
-    json account.canonical( :membership=>membership )
+    json account.canonical(membership: membership)
   end
 
   # Resend a welcome email for a pending account.
   def resend_welcome
     return forbidden unless current_account.admin?
+
     account = current_organization.accounts.find(params[:id])
-    LifecycleMailer.login_instructions( account, current_organization, current_account ).deliver_now
+    LifecycleMailer.login_instructions(account, current_organization, current_account).deliver_now
     json nil
   end
   
   def mailboxes
-    # code goes here.
+    # TODO: Filter by status, once that exists
+    @mailboxes = UploadMailbox.where(membership_id: @current_account.memberships.pluck(:id))
+    set_minimal_nav text: 'Back to workspace', link: '/search/'
   end
   
   def create_mailbox
+    @mailbox = UploadMailbox.create(membership: @current_account.memberships.default.first, sender: params[:email])
+    if @mailbox.valid?
```

Comment:
we need consistency one way or the other.  I'm not 100% whether SES enforces downcasing in email headers or not.  I'd vote for yes we should downcase everything